frame/election-provider-multi-phase/src/unsigned.rs

```diff
@@ -261,6 +268,46 @@ impl<T: Config> Pallet<T> {
 		}
 	}
 
+	/// Greedily reduce the size of the solution to fit into the block w.r.t length.
+	///
+	/// The length of the solution is largely a function of the number of voters. The number of
+	/// winners cannot be changed. Thus, to reduce the solution size, we need to strip voters.
```

Comment:
worth saying that when removing voters from a compact, you could actually end up removing a target as well.

Reply:
could we avoid removing self voter which are part of the solution ?

Reply:
We could maybe filter out self voters from the list of voters eligible to be removed, but that doesn't solve the problem in case there is a winner who (accidentally?) neglected to self-vote. We'd still need to note that the procedure can potentially strip enough voters that the number of winners changes.

Reply:
> but that doesn't solve the problem in case there is a winner who (accidentally?) neglected to self-vote.

I think staking is enforcing validator to self vote.

But in practice I wouldn't expect a trimmed solution to remove a target.
Even if we keep the last voter on the target (thus very small voter because we wanted to remove it). Then the solution score is already very very bad.

Reply:
> in practice I wouldn't expect a trimmed solution to remove a target.

Agree. The length-bounding is very much just for safety against weird cases anyways. Most of the time I would expect a solution which fits the weight bound to already be within the required length bound.